src/bootstrap/compile.rs

```diff
@@ -494,6 +500,13 @@ impl Step for Rustc {
         let compiler = self.compiler;
         let target = self.target;
 
+        if builder.config.download_rustc {
+            // Copy the existing artifacts instead of rebuilding them.
+            // NOTE: this path is only taken for tools linking to rustc-dev.
```

Comment:
Hm not sure I really follow this comment, there's nothing gating this is there? I think it'll always be the case that we skip this compile with download-rustc set?

Reply:
`download-rustc` works for everything besides tools without this block because it's handled in `impl Step for Sysroot`: https://github.com/rust-lang/rust/blob/8fb272c8e34cc3b2f7426743c6db398daba7645d/src/bootstrap/compile.rs#L923-L933
Only building rustc-dev skips the sysroot and builds rustc directly.